packages/medusa/src/api/routes/admin/discounts/create-dynamic-code.ts

```diff
@@ -55,6 +60,11 @@ export class AdminPostDiscountsDiscountDynamicCodesReq {
   @IsNotEmpty()
   code: string
 
+  @IsDate()
+  @IsOptional()
+  @Type(() => Date)
+  ends_at?: Date
+
```

Comment:
I think this should be controlled based on the `valid_duration` or  `ends_at` fields on the "parent" discount. I.e. not configurable on creation

Reply:
This must have been included by a mistake. I'll revert back 👍